py/makeqstrdata.py

```diff
@@ -100,9 +100,30 @@ def translate(translation_file, i18ns):
             translations.append((original, translation))
         return translations
 
+def frequent_ngrams(corpus, sz, n):
+    return collections.Counter(corpus[i:i+sz] for i in range(len(corpus)-sz)).most_common(n)
+
+def ngrams_to_pua(translation, ngrams):
```

Comment:
What is `pua`?

Reply:
Unicode designates a number of ranges of code points as "private use areas".  This is commonly abbreviated PUA.  The initial implementation used the private use area starting at 0xe000.  However, the current implementation uses a different region of code points, 0x80 through 0x9f, which are officially designated for other uses within Unicode but should never appear in source or translated strings in CircuitPython.  (for instance, the unicode code point 0x9f indicates APPLICATION PROGRAM COMMAND for compatibility with the ANSI Escape Code specification)

I can rename it "encode_ngrams" / "decode_ngrams" if you like.

(some of the code is specific to bigrams, other parts of the code generalize to ngrams.  This particular bit does happen to be applicable to general ngrams)

Reply:
(I went ahead and renamed them)